numpy/core/tests/test_conversion_utils.py

```diff
@@ -1,45 +1,86 @@
 """
 Tests for numpy/core/src/multiarray/conversion_utils.c
 """
+import re
+
 import pytest
+import warnings
 
 import numpy as np
 import numpy.core._multiarray_tests as mt
+from numpy.testing import suppress_warnings
 
 
 class StringConverterTestCase:
     allow_bytes = True
     case_insensitive = True
     exact_match = False
 
+    def _check_value_error(self, val):
+        with pytest.raises(ValueError) as exc:
+            self.conv(val)
+
+        assert exc.match(r'\(got {}\)'.format(re.escape(repr(val))))
+
     def _check(self, val, expected):
-        assert self.conv(val) == expected
+        # suppress deprecation warnings for case insensitive
+        # and inexact matches
+        sup = suppress_warnings()
+
+        if not self.exact_match:
+            if len(val) == 1:
+                # e.g. order
+                warn_msg = "case insensitive matches"
+            else:
+                # e.g. searchside, sortkind, clipmode
+                warn_msg = "inexact matches and case insensitive matches"
+            sup.filter(DeprecationWarning, warn_msg)
```

Comment:
Ideally you would use `pytest.warns` once for every relevant call to `self.conv`, to test _that_ it warns, not just to ignore when it warns.

Reply:
I replaced it with `assert_warns`, seems like probably due to pytest.ini settings, the DeprecationWarning still surface as errors, so pytest.warns doesnt catch it.